collector/socket/src/main/java/org/apache/karaf/decanter/collector/socket/SocketCollector.java

```diff
@@ -68,11 +74,21 @@
 
     @SuppressWarnings("unchecked")
     @Activate
-    public void activate(ComponentContext context) throws IOException {
+    public void activate(ComponentContext context) throws IOException, InvalidSyntaxException {
         this.properties = context.getProperties();
         int port = Integer.parseInt(getProperty(this.properties, "port", "34343"));
         int workers = Integer.parseInt(getProperty(this.properties, "workers", "10"));
+        this.dataformat = getProperty(this.properties, "dataformat", "json");
         
+        Collection<ServiceReference<Unmarshaller>> unmarshallers = 
+                context.getBundleContext().getServiceReferences(Unmarshaller.class, "(" +
```

Comment:
I think we can use directly `@Reference` with filter using the component properties. Let me check.

Reply:
I tried to do this but I can't retreive the @Reference filter with a property in the configuration file

Reply:
Correct, the filter can't be dynamic and use a property there. Anyway, let me do some polishing here.

Reply:
Actually, I think there's an elegant way to achieve this ;)

Let say, you have the following component:

```
@Component
public class MyComponent {
  @Reference(target="(foo=bar)")  
  OtherService otherService;
}
```
You can have the target defined in the configuration. For instance, `MyComponent` can be updated this way:

```
@Component(configurationPid="my.component")
public class MyComponent {
 @Reference
  OtherService otherService;
}
```
Then, in `my.component` configuration (`etc/my.component.cfg`), we can have the reference filter:

```
otherService.target="(foo=bar)"
```
And so, it can be updated by user.

Reply:
As discussed, to support the configuration approach, we have to update to OSGi Compendium 6.0.0 (with "new" DS/SCR version). I already prepared the update, so, please wait a bit before rebasing and using this approach.